### PR TITLE
Makefile: Fix missing brackets

### PR DESCRIPTION
**Pull request by acassis**
## Summary
These issue were found by Georgi Angelov while porting NuttX to use on PlatformIO.
## Impact
Fixes
## Testing
N/A

### CODE REVIEW

File: arch/arm/src/Makefile [hunk 1]
```diff
@@ -190,7 +190,7 @@ EXTRA_LIBS += allsyms$(OBJEXT)
 endif
 
 define LINK_ALLSYMS_KASAN
-	$(if $(CONFIG_ALLSYMS),
+	$(if $(CONFIG_ALLSYMS),)
```
Comment (xiaoxiang781216) [inline arch/arm/src/Makefile:193]: but it's the right if make expression

Comment (acassis) [reply arch/arm/src/Makefile:193]: Normally it is "$(if $(CONFIG_ALLSYMS), something, etc)", the "$(if" normally needs a closing ")"

Comment (xiaoxiang781216) [reply arch/arm/src/Makefile:193]: no, please watch ci or change your change locally.

Comment (acassis) [reply arch/arm/src/Makefile:193]: Oh ok, I saw the issue! I'll close it.

**Pull request closed without merging**